qutip/core/operators.py

```diff
@@ -767,12 +765,13 @@ def qeye_like(qobj):
         Identity operator Qobj.
 
     """
-    from .cy.qobjevo import QobjEvo
-    if isinstance(qobj, QobjEvo):
-        qobj = qobj(0)
+    if qobj.shape[0] != qobj.shape[1]:
+        raise ValueError(
+            "Can't create an identity matrix like a non square matrix."
+        )
     return Qobj(
-        _data.identity_like(qobj.data), dims=qobj._dims,
-        isherm=True, isunitary=True, copy=False
+        _data.identity[qobj.dtype](qobj.shape[0]), dims=qobj._dims,
+        isherm=True, isunitary=False, copy=False
```

Comment:
Why `isunitary=False`?

Reply:
Cut and paste error...
Fixing.
```suggestion
        isherm=True, isunitary=True, copy=False
```